mypyc/emitclass.py

```diff
@@ -199,16 +219,53 @@ def generate_constructor_for_class(cl: ClassIR,
     emitter.emit_line('}')
 
 
+def generate_constructor_for_class(cl: ClassIR,
+                                   fn: FuncIR,
+                                   init_fn: Optional[FuncIR],
+                                   setup_name: str,
+                                   vtable_name: str,
+                                   emitter: Emitter) -> None:
+    """Generate a native function that allocates and initializes an instance of a class."""
+    emitter.emit_line('{}'.format(native_function_header(fn)))
+    emitter.emit_line('{')
+    emitter.emit_line('PyObject *self = {}();'.format(setup_name))
+    emitter.emit_line('if (self == NULL)')
+    emitter.emit_line('    return NULL;')
+    if init_fn is not None:
+        args = ', '.join(['self'] + [REG_PREFIX + arg.name for arg in fn.args])
+        emitter.emit_line('{}{}({});'.format(NATIVE_PREFIX, init_fn.cname, args))
+    emitter.emit_line('return self;')
+    emitter.emit_line('}')
+
+
+def generate_init_for_class(cl: ClassIR,
+                            func_name: str,
+                            init_fn: FuncIR,
+                            emitter: Emitter) -> None:
+    """Generate an init function suitable for use as tp_init.
+
+    tp_init needs to be a function that returns an int, and our
+    __init__ methods return a PyObject. Translate NULL to -1,
+    everything else to 0."""
```

Comment:
Style nit: Trailing `"""` should be on a separate line in a multi-line docstring.